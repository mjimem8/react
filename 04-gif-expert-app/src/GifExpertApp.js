import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

export const GifExpertApp = ({defalultCategories = []}) => {

  // const [categories, setcategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball'])
  const [categories, setcategories] = useState(defalultCategories)

  // const handleAdd = () => {
  //   setcategories(cats => [...cats, 'HunterXHunter']);
  // }

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setcategories={setcategories}/>

      {/* <button onClick={handleAdd}>Agregar</button> */}
      
      <ol>
        {
          categories.map((category, i) =>
            <GifGrid key={ category } category={category} />
          )
        }
      </ol>
    </>
  )
}