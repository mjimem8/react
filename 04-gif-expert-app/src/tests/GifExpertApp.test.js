import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en el <GifExpertApp />', () => {

    test('debe de mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrarse una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defalultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    });

});