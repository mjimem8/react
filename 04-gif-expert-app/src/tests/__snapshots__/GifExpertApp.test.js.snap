// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifExpertApp /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas en el <GifExpertApp /> debe de mostrarse una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
